Binary files mythweather/i18n/mythweather_ru.qm and mythweather_new/i18n/mythweather_ru.qm differ
diff -cr mythweather/i18n/mythweather_ru.ts mythweather_new/i18n/mythweather_ru.ts
*** mythweather/i18n/mythweather_ru.ts	2009-10-04 01:34:15.000000000 +0300
--- mythweather_new/i18n/mythweather_ru.ts	2010-06-12 11:57:11.000000000 +0300
***************
*** 1,201 ****
  <?xml version="1.0" encoding="utf-8"?>
  <!DOCTYPE TS>
! <TS version="2.0">
  <context>
      <name>GlobalSetup</name>
      <message>
          <source>Finish</source>
!         <translation type="unfinished"></translation>
      </message>
  </context>
  <context>
      <name>LocationDialog</name>
      <message>
          <source>Search</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Searching ...</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Searching ... Results: %1</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Search Complete. Results: %1</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Source: %1</source>
!         <translation type="unfinished"></translation>
      </message>
  </context>
  <context>
      <name>MythControls</name>
      <message>
          <source>Weather forecasts</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Pause current page</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Search List</source>
!         <translation type="unfinished"></translation>
      </message>
  </context>
  <context>
      <name>QObject</name>
      <message>
          <source>Current Conditions</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Three Day Forecast</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>18 Hour Forecast</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Severe Weather Alerts</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Six Day Forecast</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Static Map</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Animated Map</source>
!         <translation type="unfinished"></translation>
      </message>
  </context>
  <context>
      <name>ScreenSetup</name>
      <message>
          <source>Active Screens</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Inactive Screens</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Finish</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Add desired screen to the Active Screens list by pressing SELECT.</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Sources</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Units: </source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>English Units</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>SI Units</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Location: </source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Not Defined</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Source: </source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Press SELECT to </source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>change location; </source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>change units; </source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>move screen up or down; or remove screen.</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Manipulate Screen</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Move Up</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Move Down</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Remove</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Change Location</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Change Units</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Cancel</source>
!         <translation type="unfinished"></translation>
      </message>
  </context>
  <context>
      <name>SourceManager</name>
      <message>
          <source>Searching for scripts</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Script %1 is still running when trying to do update, Make sure it isn&apos;t hanging, make sure timeout values are sane... Not running this time around</source>
!         <translation type="unfinished"></translation>
      </message>
  </context>
  <context>
      <name>SourceSetup</name>
      <message>
          <source>Finish</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Author: </source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Email: </source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Version: </source>
!         <translation type="unfinished"></translation>
      </message>
  </context>
  <context>
--- 1,201 ----
  <?xml version="1.0" encoding="utf-8"?>
  <!DOCTYPE TS>
! <TS version="2.0" language="ru_RU">
  <context>
      <name>GlobalSetup</name>
      <message>
          <source>Finish</source>
!         <translation>Закончить</translation>
      </message>
  </context>
  <context>
      <name>LocationDialog</name>
      <message>
          <source>Search</source>
!         <translation>Искать</translation>
      </message>
      <message>
          <source>Searching ...</source>
!         <translation>Поиск ...</translation>
      </message>
      <message>
          <source>Searching ... Results: %1</source>
!         <translation>Поиск ... Найдено: %1</translation>
      </message>
      <message>
          <source>Search Complete. Results: %1</source>
!         <translation>Поиск закончен. Найдено:%1</translation>
      </message>
      <message>
          <source>Source: %1</source>
!         <translation>Источник: %1</translation>
      </message>
  </context>
  <context>
      <name>MythControls</name>
      <message>
          <source>Weather forecasts</source>
!         <translation>Прогноз погоды</translation>
      </message>
      <message>
          <source>Pause current page</source>
!         <translation>Остановить экран</translation>
      </message>
      <message>
          <source>Search List</source>
!         <translation>Список поиска</translation>
      </message>
  </context>
  <context>
      <name>QObject</name>
      <message>
          <source>Current Conditions</source>
!         <translation>Текущее состояние</translation>
      </message>
      <message>
          <source>Three Day Forecast</source>
!         <translation>Прогноз на три дня</translation>
      </message>
      <message>
          <source>18 Hour Forecast</source>
!         <translation>Прогноз на 18 часов</translation>
      </message>
      <message>
          <source>Severe Weather Alerts</source>
!         <translation>Оповещение о стихийних бедствиях</translation>
      </message>
      <message>
          <source>Six Day Forecast</source>
!         <translation>Погода на шесть дней</translation>
      </message>
      <message>
          <source>Static Map</source>
!         <translation>Карта погоды</translation>
      </message>
      <message>
          <source>Animated Map</source>
!         <translation>Анимированная карта</translation>
      </message>
  </context>
  <context>
      <name>ScreenSetup</name>
      <message>
          <source>Active Screens</source>
!         <translation>Активные экраны</translation>
      </message>
      <message>
          <source>Inactive Screens</source>
!         <translation>Не активные экраны</translation>
      </message>
      <message>
          <source>Finish</source>
!         <translation>Закончить</translation>
      </message>
      <message>
          <source>Add desired screen to the Active Screens list by pressing SELECT.</source>
!         <translation>Для добавления экрана к списку активных нажмите SELECT.</translation>
      </message>
      <message>
          <source>Sources</source>
!         <translation>Источники</translation>
      </message>
      <message>
          <source>Units: </source>
!         <translation>Единицы:</translation>
      </message>
      <message>
          <source>English Units</source>
!         <translation>Имперская система</translation>
      </message>
      <message>
          <source>SI Units</source>
!         <translation>Метрическая система</translation>
      </message>
      <message>
          <source>Location: </source>
!         <translation>Расположение:</translation>
      </message>
      <message>
          <source>Not Defined</source>
!         <translation>Не определено</translation>
      </message>
      <message>
          <source>Source: </source>
!         <translation>Источник:</translation>
      </message>
      <message>
          <source>Press SELECT to </source>
!         <translation>Нажмите SELECT для </translation>
      </message>
      <message>
          <source>change location; </source>
!         <translation>изменить расположение;</translation>
      </message>
      <message>
          <source>change units; </source>
!         <translation>изменить систему измерения;</translation>
      </message>
      <message>
          <source>move screen up or down; or remove screen.</source>
!         <translation>сдвинуть экран вверх или вниз; или удалить экран.</translation>
      </message>
      <message>
          <source>Manipulate Screen</source>
!         <translation>Управление экраном</translation>
      </message>
      <message>
          <source>Move Up</source>
!         <translation>Сдвинуть вверх</translation>
      </message>
      <message>
          <source>Move Down</source>
!         <translation>Сдвинуть вниз</translation>
      </message>
      <message>
          <source>Remove</source>
!         <translation>Удалить</translation>
      </message>
      <message>
          <source>Change Location</source>
!         <translation>Изменить расположение</translation>
      </message>
      <message>
          <source>Change Units</source>
!         <translation>Изменить систему измерения</translation>
      </message>
      <message>
          <source>Cancel</source>
!         <translation>Отменить</translation>
      </message>
  </context>
  <context>
      <name>SourceManager</name>
      <message>
          <source>Searching for scripts</source>
!         <translation>Поиск скриптов</translation>
      </message>
      <message>
          <source>Script %1 is still running when trying to do update, Make sure it isn&apos;t hanging, make sure timeout values are sane... Not running this time around</source>
!         <translation>Скрипт %1 запущен и пытается обновить данные. Убедитесь что он работает и интервал обновления правильный ... Не запускайте еще раз</translation>
      </message>
  </context>
  <context>
      <name>SourceSetup</name>
      <message>
          <source>Finish</source>
!         <translation>Закончить</translation>
      </message>
      <message>
          <source>Author: </source>
!         <translation>Автор:</translation>
      </message>
      <message>
          <source>Email: </source>
!         <translation>Email:</translation>
      </message>
      <message>
          <source>Version: </source>
!         <translation>Версия:</translation>
      </message>
  </context>
  <context>
***************
*** 747,859 ****
      </message>
      <message>
          <source>Wind(Gust)</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Wind Direction</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Feels Like</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>High</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Low</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Global Settings</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Seconds to display each page:</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Retrieve data in the background:</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Source Setup</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Data Retrieval Interval (minutes):</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Source Update Timeout (seconds):</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Weather Alerts</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Temp</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Precip</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Screen Setup</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Location Search</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Enter your current location, nearest large town or city. In some countries postal codes are also supported.</source>
!         <translation type="unfinished"></translation>
      </message>
      <message>
          <source>Enter Location:</source>
!         <translation type="unfinished"></translation>
      </message>
  </context>
  <context>
      <name>Weather</name>
      <message>
          <source>Paused</source>
!         <translation type="unfinished"></translation>
      </message>
  </context>
  <context>
      <name>WeatherScreen</name>
      <message>
          <source>Sunday</source>
!         <translation type="unfinished">Воскресенье</translation>
      </message>
      <message>
          <source>Monday</source>
!         <translation type="unfinished">Понедельник</translation>
      </message>
      <message>
          <source>Tuesday</source>
!         <translation type="unfinished">Вторник</translation>
      </message>
      <message>
          <source>Wednesday</source>
!         <translation type="unfinished">Среда</translation>
      </message>
      <message>
          <source>Thursday</source>
!         <translation type="unfinished">Четверг</translation>
      </message>
      <message>
          <source>Friday</source>
!         <translation type="unfinished">Пятница</translation>
      </message>
      <message>
          <source>Saturday</source>
!         <translation type="unfinished">Субота</translation>
      </message>
  </context>
  </TS>
--- 747,859 ----
      </message>
      <message>
          <source>Wind(Gust)</source>
!         <translation>Скорость ветра</translation>
      </message>
      <message>
          <source>Wind Direction</source>
!         <translation>Направление ветра</translation>
      </message>
      <message>
          <source>Feels Like</source>
!         <translation>По ощущению</translation>
      </message>
      <message>
          <source>High</source>
!         <translation>Днем</translation>
      </message>
      <message>
          <source>Low</source>
!         <translation>Ночью</translation>
      </message>
      <message>
          <source>Global Settings</source>
!         <translation>Общие настройки</translation>
      </message>
      <message>
          <source>Seconds to display each page:</source>
!         <translation>Время показа каждой страницы (сек.):</translation>
      </message>
      <message>
          <source>Retrieve data in the background:</source>
!         <translation>Получать данные в фоне:</translation>
      </message>
      <message>
          <source>Source Setup</source>
!         <translation>Настройка источника</translation>
      </message>
      <message>
          <source>Data Retrieval Interval (minutes):</source>
!         <translation>Интервал получения данных (мин.):</translation>
      </message>
      <message>
          <source>Source Update Timeout (seconds):</source>
!         <translation>Таймаут обновления (сек.):</translation>
      </message>
      <message>
          <source>Weather Alerts</source>
!         <translation>Предупреждения</translation>
      </message>
      <message>
          <source>Temp</source>
!         <translation>Темп</translation>
      </message>
      <message>
          <source>Precip</source>
!         <translation>Осадки</translation>
      </message>
      <message>
          <source>Screen Setup</source>
!         <translation>Настройки экрана</translation>
      </message>
      <message>
          <source>Location Search</source>
!         <translation>Поиск местоположения</translation>
      </message>
      <message>
          <source>Enter your current location, nearest large town or city. In some countries postal codes are also supported.</source>
!         <translation>Введите ваше текущее расположение (город поблизости). Для некоторых стран поддерживается почтовый индекс.</translation>
      </message>
      <message>
          <source>Enter Location:</source>
!         <translation>Введите месторасположение:</translation>
      </message>
  </context>
  <context>
      <name>Weather</name>
      <message>
          <source>Paused</source>
!         <translation>Остановлено</translation>
      </message>
  </context>
  <context>
      <name>WeatherScreen</name>
      <message>
          <source>Sunday</source>
!         <translation>Воскресенье</translation>
      </message>
      <message>
          <source>Monday</source>
!         <translation>Понедельник</translation>
      </message>
      <message>
          <source>Tuesday</source>
!         <translation>Вторник</translation>
      </message>
      <message>
          <source>Wednesday</source>
!         <translation>Среда</translation>
      </message>
      <message>
          <source>Thursday</source>
!         <translation>Четверг</translation>
      </message>
      <message>
          <source>Friday</source>
!         <translation>Пятница</translation>
      </message>
      <message>
          <source>Saturday</source>
!         <translation>Суббота</translation>
      </message>
  </context>
  </TS>
diff -cr mythweather/mythweather/dbcheck.cpp mythweather_new/mythweather/dbcheck.cpp
*** mythweather/mythweather/dbcheck.cpp	2009-09-21 19:28:05.000000000 +0300
--- mythweather_new/mythweather/dbcheck.cpp	2010-06-13 22:34:02.000000000 +0300
***************
*** 8,14 ****
  
  #include "dbcheck.h"
  
! const QString currentDatabaseVersion = "1004";
  
  static bool UpdateDBVersionNumber(const QString &newnumber)
  {
--- 8,14 ----
  
  #include "dbcheck.h"
  
! const QString currentDatabaseVersion = "1005";
  
  static bool UpdateDBVersionNumber(const QString &newnumber)
  {
***************
*** 194,198 ****
              return false;
      }
  
      return true;
! }
--- 194,208 ----
              return false;
      }
  
+     if (dbver == "1004") 
+    { 
+ 	    QStringList updates; 
+  	    updates << "ALTER TABLE weatherdatalayout" 
+  	                   "  MODIFY location varchar(128) CHARACTER SET utf8 NOT NULL;"; 
+ 	
+ 	    if (!performActualUpdate(updates, "1005", dbver)) 
+ 		return false; 
+     }
+     
      return true;
! }
\ No newline at end of file
diff -cr mythweather/mythweather/scripts/README mythweather_new/mythweather/scripts/README
*** mythweather/mythweather/scripts/README	2009-09-29 23:06:06.000000000 +0300
--- mythweather_new/mythweather/scripts/README	2010-06-13 22:34:00.000000000 +0300
***************
*** 8,17 ****
         script -u units [-d dir] location
  
  -v -- Information about the script in format
! name,version,author,email
  
  -T -- default timeout values specified by this script in form
!         updateTimeout,retieveTimeout
          updateTimeout -- time between updates
          retrieveTimeout -- timeout when trying to retrieve data
  
--- 8,17 ----
         script -u units [-d dir] location
  
  -v -- Information about the script in format
! 	name,version,author,email
  
  -T -- default timeout values specified by this script in form
!         updateTimeout,retrieveTimeout
          updateTimeout -- time between updates
          retrieveTimeout -- timeout when trying to retrieve data
  
***************
*** 42,48 ****
  
  item::This is a 
  item::multi-line 
! itme::data item
  
  These must be printed successively, behavior is undefined otherwise.
  
--- 42,48 ----
  
  item::This is a 
  item::multi-line 
! item::data item
  
  These must be printed successively, behavior is undefined otherwise.
  
***************
*** 60,74 ****
  setup screen the scripts directory is searched for scripts.  New scripts are
  added to the database and existing scripts are checked for version changes.
  
! Types are the screen items as defined in weather-ui.xml.  A data item with no
! modifiers before it is a static item and the plugin will not search for a script
! to provide data to it. A * signifies an item that is dynamic, but not required
! to display the screen, these are not provided by scripts, they are things like
! pause text or the current date and time.  A + prepending the item name signifies
! an item that is required to show the screen, these are provided by scripts.  The
! text of the name within the weather-ui.xml file (with prepending flag removed)
! must match exactly the string returned by the '-t' flag; as well as the token
! returned when retrieving data.
  
  If you want to see the code that controls the scripts within the plugin, look at
  weatherSource.cpp and sourceManager.cpp.  If you have any problems with the
--- 60,69 ----
  setup screen the scripts directory is searched for scripts.  New scripts are
  added to the database and existing scripts are checked for version changes.
  
! Types are the screen items as defined in weather-screens.xml. 
! The text of the name within the weather-screens.xml file must match exactly the 
! string returned by the '-t' flag; as well as the token returned when retrieving  
! data. 
  
  If you want to see the code that controls the scripts within the plugin, look at
  weatherSource.cpp and sourceManager.cpp.  If you have any problems with the
diff -cr mythweather/mythweather/weatherScreen.cpp mythweather_new/mythweather/weatherScreen.cpp
*** mythweather/mythweather/weatherScreen.cpp	2009-08-25 21:12:13.000000000 +0300
--- mythweather_new/mythweather/weatherScreen.cpp	2010-06-13 23:05:40.000000000 +0300
***************
*** 161,173 ****
          return value + " %";
  
      if (key == "pressure")
!         return value + (m_units == ENG_UNITS ? " in" : " mb");
  
      if (key == "visibility")
          return value + (m_units == ENG_UNITS ? " mi" : " km");
  
      if (key == "temp" || key == "appt" || key.contains("low",Qt::CaseInsensitive) ||
!         key.contains("high",Qt::CaseInsensitive))
      {
         if ( (value == "NA") || (value == "N/A") )
            return value;
--- 161,174 ----
          return value + " %";
  
      if (key == "pressure")
!         return value + (m_units == ENG_UNITS ? " in" : " mmHg");
  
      if (key == "visibility")
          return value + (m_units == ENG_UNITS ? " mi" : " km");
  
      if (key == "temp" || key == "appt" || key.contains("low",Qt::CaseInsensitive) ||
!         key.contains("high",Qt::CaseInsensitive) ||
!         key.contains("temp",Qt::CaseInsensitive))
      {
         if ( (value == "NA") || (value == "N/A") )
            return value;
***************
*** 176,182 ****
      }
  
      if (key == "wind_gust" || key == "wind_spdgst" || key == "wind_speed")
!         return value + (m_units == ENG_UNITS ? " mph" : " kph");
  
      /*The days of the week will be translated if the script sends elements from
       the enum DaysOfWeek.*/
--- 177,183 ----
      }
  
      if (key == "wind_gust" || key == "wind_spdgst" || key == "wind_speed")
!         return value + (m_units == ENG_UNITS ? " mph" : " m/s");
  
      /*The days of the week will be translated if the script sends elements from
       the enum DaysOfWeek.*/
diff -cr mythweather/mythweather/weather-screens.xml mythweather_new/mythweather/weather-screens.xml
*** mythweather/mythweather/weather-screens.xml	2009-08-22 23:38:13.000000000 +0300
--- mythweather_new/mythweather/weather-screens.xml	2010-06-13 22:34:04.000000000 +0300
***************
*** 32,37 ****
--- 32,38 ----
          <datum name="low-2" />
          <datum name="high-2" />
          <datum name="updatetime" />
+ 	<datum name="copyright" />
      </screen>
  
      <screen name="Six Day Forecast">
***************
*** 61,89 ****
--- 62,95 ----
          <datum name="low-4" />
          <datum name="low-5" />
          <datum name="updatetime" />
+ 	<datum name="copyright" />
      </screen>
  
      <!-- <screen name="Today's Forecast">
          <datum name="header" />
          <datum name="updatetime" />
          <datum name="today-desc" />
+ 	<datum name="copyright" />
      </screen> -->
  
      <screen name="Static Map" hasunits="no">
          <datum name="smdesc" />
          <datum name="updatetime" />
          <datum name="map" />
+ 	<datum name="copyright" />
      </screen>
  
      <screen name="Animated Map" hasunits="no">
          <datum name="amdesc" />
          <datum name="updatetime" />
          <datum name="animatedimage" />
+ 	<datum name="copyright" />
      </screen>
  
      <screen name="Severe Weather Alerts" hasunits="no">
          <datum name="swlocation" />
          <datum name="alerts" />
+ 	<datum name="copyright" />
      </screen>
  
      <screen name="18 Hour Forecast">
***************
*** 113,118 ****
--- 119,125 ----
          <datum name="pop-4" />
          <datum name="pop-5" />
          <datum name="updatetime" />
+ 	<datum name="copyright" />
      </screen>
  
      <!-- <screen name="Tomorrow's Outlook">
***************
*** 122,126 ****
--- 129,134 ----
          <datum name="todaypic" />
          <datum name="updatetime" />
          <datum name="location" />
+ 	<datum name="copyright" />
      </screen> -->
  </screens>
diff -cr mythweather/mythweather/weatherSource.cpp mythweather_new/mythweather/weatherSource.cpp
*** mythweather/mythweather/weatherSource.cpp	2010-03-13 21:09:32.000000000 +0200
--- mythweather_new/mythweather/weatherSource.cpp	2010-06-13 22:34:08.000000000 +0300
***************
*** 4,9 ****
--- 4,10 ----
  #include <QDir>
  #include <QFile>
  #include <QTextStream>
+ #include <QTextCodec> 
  #include <QApplication>
  
  // MythTV headers
***************
*** 475,480 ****
--- 476,483 ----
  
      QStringList locs;
      m_proc->setReadChannel(QProcess::StandardOutput);
+     
+     QTextCodec *codec = QTextCodec::codecForName("UTF-8");
      while (m_proc->canReadLine())
      {
          QByteArray tmp = m_proc->readLine();
***************
*** 483,489 ****
              tmp.chop(1);
  
          if (!tmp.isEmpty())
!             locs.push_back(QString(tmp));
      }
  
      return locs;
--- 486,495 ----
              tmp.chop(1);
  
          if (!tmp.isEmpty())
! 	{
! 	    QString loc_string = codec->toUnicode(tmp);
! 	     locs.push_back(loc_string);	
! 	}
      }
  
      return locs;
***************
*** 596,602 ****
      if (!tempStr.isEmpty())
          m_buffer += tempStr;
  
!     QString cachefile = QString("%1/cache_%2").arg(m_dir).arg(m_locale);
      QFile cache(cachefile);
      if (cache.open( QIODevice::WriteOnly ))
      {
--- 602,609 ----
      if (!tempStr.isEmpty())
          m_buffer += tempStr;
  
!     QString locale_file = m_locale.replace("/", "-");
!     QString cachefile = QString("%1/cache_%2").arg(m_dir).arg(locale_file);
      QFile cache(cachefile);
      if (cache.open( QIODevice::WriteOnly ))
      {
***************
*** 631,637 ****
  
  void WeatherSource::processData()
  {
!     QStringList data = QString(m_buffer).split('\n', QString::SkipEmptyParts);
  
      for (int i = 0; i < data.size(); ++i)
      {
--- 638,646 ----
  
  void WeatherSource::processData()
  {
!     QTextCodec *codec = QTextCodec::codecForName("UTF-8");
!     QString unicode_buffer = codec->toUnicode(m_buffer); 
!     QStringList data = unicode_buffer.split('\n', QString::SkipEmptyParts);	
  
      for (int i = 0; i < data.size(); ++i)
      {
***************
*** 647,658 ****
              return; // we don't emit signal
          }
  
!         if (!m_data[temp[0]].isEmpty())
!         {
!             m_data[temp[0]].append("\n" + temp[1]);
!         }
!         else
!             m_data[temp[0]] = temp[1];
      }
  }
  
--- 656,670 ----
              return; // we don't emit signal
          }
  
! 	if (temp[1] != "---") 
! 	{ 
!  	    if (!m_data[temp[0]].isEmpty()) 
!  	   { 
!  	        m_data[temp[0]].append("\n" + temp[1]); 
!  	   } 
! 	   else 
! 		m_data[temp[0]] = temp[1]; 
!  	} 
      }
  }
  
